lib/livebook_web/live/file_select_component.ex

```diff
@@ -70,7 +71,14 @@ defmodule LivebookWeb.FileSelectComponent do
       |> assign(assigns)
       |> update_file_infos(force_reload? or running_files_changed?)
 
-    {:ok, socket}
+    configure_path = ~p"/hub/#{Livebook.Hubs.Personal.id()}/file-systems/new"
+
+    {file_systems, configure_path} =
+      if hub = socket.assigns[:hub],
+        do: {Livebook.Hubs.get_file_systems(hub), ~p"/hub/#{hub.id}/file-systems/new"},
+        else: {Livebook.Hubs.get_file_systems(), configure_path}
```

Comment:
```suggestion
    {file_systems, configure_path} =
      if hub = socket.assigns[:hub],
        do: {Livebook.Hubs.get_file_systems(hub), ~p"/hub/#{hub.id}/file-systems/new"},
        else: {Livebook.Hubs.get_file_systems(), ~p"/hub/#{Livebook.Hubs.Personal.id()}/file-systems/new"}
```

I wonder though if we should always explicitly pass those as arguments. @jonatanklosko, any preference?

Reply:
@josevalim no strong preference, these are always a function of what (if any) hub we use for, so I think it's fine to encapsulate the logic here, but keeping it more generic is also fine.